test/order_test.rb

```diff
@@ -52,84 +54,122 @@
 
-    it "Returns a total of zero if there are no products" do
-      order = Order.new(1337, {}, customer)
+      it "Returns a total of zero if there are no products" do
+        order = Order.new(1337, {}, customer)
 
-      expect(order.total).must_equal 0
+        expect(order.total).must_equal 0
+      end
     end
-  end
 
-  describe "#add_product" do
-    it "Increases the number of products" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
-      before_count = products.count
-      order = Order.new(1337, products, customer)
+    describe "#add_product" do
+      it "Increases the number of products" do
+        products = { "banana" => 1.99, "cracker" => 3.00 }
+        before_count = products.count
+        order = Order.new(1337, products, customer)
 
-      order.add_product("salad", 4.25)
-      expected_count = before_count + 1
-      expect(order.products.count).must_equal expected_count
-    end
+        order.add_product("salad", 4.25)
+        expected_count = before_count + 1
+        expect(order.products.count).must_equal expected_count
+      end
 
-    it "Is added to the collection of products" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
-      order = Order.new(1337, products, customer)
+      it "Is added to the collection of products" do
+        products = { "banana" => 1.99, "cracker" => 3.00 }
+        order = Order.new(1337, products, customer)
 
-      order.add_product("sandwich", 4.25)
-      expect(order.products.include?("sandwich")).must_equal true
+        order.add_product("sandwich", 4.25)
+        expect(order.products.include?("sandwich")).must_equal true
+      end
+
+      it "Raises an ArgumentError if the product is already present" do
+        products = { "banana" => 1.99, "cracker" => 3.00 }
+
+        order = Order.new(1337, products, customer)
+        before_total = order.total
+
+        expect {
+          order.add_product("banana", 4.25)
+        }.must_raise ArgumentError
+
+        # the list of products should not have been modified
+        expect(order.total).must_equal before_total
+      end
     end
+    # remove method
+    describe "#remove_product" do
+      it "decreases the number of products" do
+        products = { "banana" => 1.99, "cracker" => 3.00 }
+        before_count = products.count
+        order = Order.new(1563, products, customer)
+
+        result = order.remove_product("banana")
+        expected_count = before_count - 1
+        expect(result.count).must_equal expected_count
+      end
 
-    it "Raises an ArgumentError if the product is already present" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
+      it "Is removed from the collection of products" do
+        products = { "banana" => 1.99, "cracker" => 3.00 }
+        order = Order.new(1563, products, customer)
 
-      order = Order.new(1337, products, customer)
-      before_total = order.total
+        shopping_bag = order.remove_product("banana")
```

Comment:
You never used `shopping_bag`.

I noticed this because when you run `rake` you get this warning:
```
/Users/kaida/Developer/ada/cohort-13/grocery-store/test/order_test.rb:131: warning: assigned but unused variable - shopping_bag
```

Lots of the warnings Ruby generates can be really helpful and will sometimes catch bugs.